### PR TITLE
Enrich self register labels for runtimes

### PR DESCRIPTION
**Pull request by PetarTodorovv**
**Description**
Enrich the labels that we return during self-registration of runtimes

Changes proposed in this pull request:
- return the SaaS application name as part of the self-registered runtime's labels
- Make the new label immutable
- Adjust unit and E2E tests

**Pull Request status**
- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] [N/A] Mocks are regenerated, using the automated script

### CODE REVIEW

File: tests/director/tests/runtime_api_test.go [hunk 1]
```diff
@@ -798,53 +798,65 @@ func TestSelfRegMoreThanOneProviderRuntime(t *testing.T) {
 
 func TestRuntimeTypeImmutability(t *testing.T) {
```
Comment (DimitarPetrov) [inline tests/director/tests/runtime_api_test.go:799]: Add assert that the new label is added to the self reg test as well.

File: components/director/pkg/config/selfreg_config.go [hunk 2]
```diff
@@ -68,3 +86,24 @@ func (c *SelfRegConfig) MapInstanceConfigs() error {
...
+	if err != nil {
+		return errors.Wrapf(err, "while getting SaaS application names secret")
+	}
+
+	m, err := ParseConfigToJSONMap(secretData)
+	if err != nil {
+		return err
+	}
+
+	c.RegionToSaaSAppName = make(map[string]string)
```
Comment (la4ezar) [inline components/director/pkg/config/selfreg_config.go:102]: You can create this map with capacity = len(m).

**Review by la4ezar — CHANGES_REQUESTED** [review]

**Review by la4ezar — APPROVED** [review]

**Review by la4ezar — APPROVED** [review]

**Pull request merged**